common/dpcpp/cuckoo_hashtable.hpp

```diff
@@ -34,31 +36,24 @@ template <class Key, class Val, class Hasher1, class Hasher2> class CuckooHashta
         }
 
         bool insert(Key key, Val value) {
-            // TODO: change loop detection
-            size_t cnt = 0;
-            while (cnt < _size) {
-                size_t pos[] = {_hasher1(key), _hasher2(key)};
+            size_t pos = _hasher1(key);
+            for (int cnt = 0; cnt < std::min(_size / 4, (size_t) 1000000); cnt++) {
```

Comment:
Please get rid of magic numbers, use named compile time constants instead

Reply:
done